src/routes/ArticleCard.jsx

```diff
@@ -25,10 +25,48 @@ const ArticleCard = ({ handleArticlesClick }) => {
     fetchArticle();
   }, [article_id]);
 
+  const handleVote = (voteType) => {
+    const voteValue = voteType === "up" ? 1 : -1;
+    setArticle((prevArticle) => ({
+      ...prevArticle,
+      votes: prevArticle.votes + voteValue,
+    }));
+
+    updateArticleVotes(article_id, voteValue)
+      // change voteValue to "nonsense" for the error feedback
+      .then(() => {
+        setArticle((prevArticle) => ({
+          ...prevArticle,
+          votes: prevArticle.votes,
+          voteSuccess: true,
+        }));
+        setTimeout(() => {
+          setArticle((prevArticle) => ({
+            ...prevArticle,
+            voteSuccess: false,
+          }));
+        }, 300);
+      })
+      .catch((error) => {
+        alert("<<< Failed to update the article vote >>>");
```

Comment:
Just to note, when you use alert, it blocks the execution of any code so your app will freeze